sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -869,6 +869,14 @@ class DDLSuite extends QueryTest with SharedSQLContext with BeforeAndAfterEach {
   }
 
   test("alter table: rename partition") {
+    testRenamePartition(isDatasourceTable = false)
+  }
+
+  test("alter table: rename partition (datasource table)") {
+    testRenamePartition(isDatasourceTable = true)
+  }
+
+  private def testRenamePartition(isDatasourceTable: Boolean): Unit = {
```

Comment:
I'd like to create a method to set up environment to test rename partition, instead of creating a common testing method and pass in a flag.